internal/tui/runner.go

```diff
@@ -0,0 +1,293 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package tui
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+// questionnaireModel handles multiple questions
+type questionnaireModel struct {
+	questions       []*Question
+	currentQuestion int
+	answers         map[string]interface{}
+	finished        bool
+	err             error
+	width           int
+	height          int
+}
+
+func newQuestionnaireModel(questions []*Question) *questionnaireModel {
+	return &questionnaireModel{
+		questions: questions,
+		answers:   make(map[string]interface{}),
+		width:     80,
+		height:    24,
+	}
+}
+
+func (m *questionnaireModel) Init() tea.Cmd {
+	return nil
+}
+
+func (m *questionnaireModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		m.width = msg.Width
+		m.height = msg.Height
+		return m, nil
+
+	case tea.KeyMsg:
+		if m.finished {
+			return m, tea.Quit
+		}
+
+		switch msg.String() {
+		case "ctrl+c":
+			m.err = fmt.Errorf("cancelled by user")
+			return m, tea.Quit
+
+		case "enter":
+			if err := m.validateCurrentAnswer(); err != nil {
+				m.setCurrentError(err.Error())
+				return m, nil
+			}
+
+			// Save current answer
+			current := m.questions[m.currentQuestion]
+			m.answers[current.Name] = current.Prompt.Value()
+
+			// Move to next question or finish
+			m.currentQuestion++
+			if m.currentQuestion >= len(m.questions) {
+				m.finished = true
+				return m, tea.Quit
+			}
+
+			// Clear any previous error
+			m.setCurrentError("")
+			return m, nil
+		}
+	}
+
+	// Update current prompt
+	if !m.finished && m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		updatedPrompt, cmd := current.Prompt.Update(msg)
+		current.Prompt = updatedPrompt
+		return m, cmd
+	}
+
+	return m, nil
+}
+
+func (m *questionnaireModel) validateCurrentAnswer() error {
+	current := m.questions[m.currentQuestion]
+	if current.Validate != nil {
+		return current.Validate(current.Prompt.Value())
+	}
+	return nil
+}
+
+func (m *questionnaireModel) setCurrentError(err string) {
+	if m.currentQuestion < len(m.questions) {
+		current := m.questions[m.currentQuestion]
+		switch p := current.Prompt.(type) {
+		case interface{ SetError(string) }:
+			p.SetError(err)
+		}
+	}
+}
+
+func (m *questionnaireModel) View() string {
+	if m.finished {
+		return ""
+	}
+
+	var b strings.Builder
+
+	// Header with progress
+	progress := fmt.Sprintf("Question %d of %d", m.currentQuestion+1, len(m.questions))
```

Comment:
This is nice, but in some cases we ask at different rounds, so the total count is not known beforehand. For example when creating data streams we add a question if the user selected the logs type on the first round of questions.

With this code after "Question 4 of 4" there is a "Question 1 of 1".

As the final number of questions can be unpredictable, I wonder if it should be shown :thinking:

Or I wonder if there is some way to skip questions depending on the value of previous questions, then maybe we can represent the status better in these cases.